src/explore-education-statistics-frontend/src/modules/table-tool/components/TableToolShare.tsx

```diff
@@ -16,6 +16,7 @@ import React, { useEffect, useState } from 'react';
 
 const linkInstructions =
   'Use the link below to see a version of this page that you can bookmark for future reference, or copy the link to send on to somebody else to view.';
+const permalinkBaseUrl = `${window.location.origin}/data-tables/permalink`;
```

Comment:
This seems like it could cause SSR issues as `window` isn't necessarily available in module scope.

Would suggest moving this into the component with state. We could also clean this up a little whilst we're here:

```tsx
const TableToolShare = (...) => {
  const [permalinkUrl, setPermalinkUrl] = useState(''); // replaces permalinkId
  
  ...

  const handlePermalinkClick = async () => {
    ...
    
    setPermalinkUrl(`${window.location.origin}/data-tables/permalink/${id}`);
  };

  const handleCopyClick = () => {
    navigator.clipboard.writeText(permalinkUrl);
  };

  ...
}
```

Note that the link for viewing the permalink should also use the `permalinkUrl` variable

Reply:
doh! yes, good catch. That's sorted and a test added now.